build-logic/conventions/src/main/kotlin/com/squareup/anvil/conventions/BasePlugin.kt

```diff
@@ -0,0 +1,154 @@
+   * See https://github.com/square/anvil/pull/763#discussion_r1379563691
+   */
+  private fun Project.copyRootProjectGradleProperties() {
+    rootProject.file("../../gradle.properties")
+      .inputStream()
+      .use { Properties().apply { load(it) } }
+      .forEach { key, value ->
+        extras.set(key.toString(), value.toString())
+      }
+  }
+
+  private fun configureGradleProperties(target: Project) {
+    target.version = target.property("VERSION_NAME") as String
+    target.group = target.property("GROUP") as String
+  }
+
+  private fun configureKotlin(target: Project) {
+    target.tasks.withType(KotlinCompile::class.java).configureEach { task ->
+      task.compilerOptions {
+        allWarningsAsErrors.set(target.libs.versions.config.warningsAsErrors.get().toBoolean())
+
+        // Only add the experimental opt-in if the project has the `annotations` dependency,
+        // otherwise the compiler will throw a warning and fail in CI.
+        if (target.hasAnnotationDependency()) {
+          freeCompilerArgs.add("-opt-in=com.squareup.anvil.annotations.ExperimentalAnvilApi")
+        }
+
+        val fromInt = when (val targetInt = target.jvmTargetInt()) {
+          8 -> JVM_1_8
+          else -> JvmTarget.fromTarget("$targetInt")
+        }
+        jvmTarget.set(fromInt)
+      }
+    }
+  }
+
+  /**
+   * This is an imperfect but pretty good heuristic
+   * to determine if the receiver has the `annotations` dependency,
+   * without actually resolving the dependency graph.
+   */
+  private fun Project.hasAnnotationDependency(): Boolean {
+    val seed = sequenceOf(
+      "compileClasspath",
+      "testCompileClasspath",
+      "debugCompileClasspath",
+    )
+      .mapNotNull { configurations.findByName(it) }
+
+    val configs = generateSequence(seed) { configs ->
+      configs.flatMap { it.extendsFrom }
+        .mapNotNull { configurations.findByName(it.name) }
+        .takeIf { it.iterator().hasNext() }
+    }
+      .flatten()
+      .distinct()
+
+    // The -api and -utils projects declare the annotations as an `api` dependency.
+    val providingProjects = setOf("annotations", "compiler-api", "compiler-utils")
+
+    return configs.any { cfg ->
+      cfg.dependencies.any { dep ->
+
+        dep.group == "com.squareup.anvil" && dep.name in providingProjects
+      }
+    }
+  }
+
+  private fun configureJava(target: Project) {
+    target.plugins.withId("java") {
+      target.javaExtension.toolchain {
+        it.languageVersion.set(JavaLanguageVersion.of(target.libs.versions.jvm.toolchain.get()))
+      }
+      target.tasks.withType(JavaCompile::class.java).configureEach { task ->
+        task.options.release.set(target.jvmTargetInt())
+      }
+    }
+  }
+
+  private fun configureTests(target: Project) {
+    target.tasks.withType(Test::class.java).configureEach { task ->
+
+      task.maxParallelForks = Runtime.getRuntime().availableProcessors()
```

Comment:
It looks like previously we only did this for local machines. Is there any downside to now setting it on CI?

Reply:
Nope.  It's a pretty common setting.